### ISSUE TITLE
Make it possible to write scripts with shebang #! line

### ISSUE THREAD

**Issue by jhidding** [entangled/entangled#63]
Currently the first line in the code block ends up third in the program output. Maybe there's a way to add support for setting a header area that goes before the first entangled comment. The following input would then give:

~~~markdown
The most useless program

``` {.python file=myscript header=1}
#!/usr/bin/env python3

import sys

sys.exit(0)
```
~~~

generating the file `myscript`

~~~python
#!/usr/bin/env python3
# ~\~ language=Python filename=myscript header=1
# ~\~ begin <<lit/index.md|myscript>>[0]

import sys

sys.exit(0)
# ~\~ end
~~~

The parser can see that the first line belongs to the `#myscript` block.

**Comment by MyriaCore** [comment]
I'll add that there's no reason why this couldn't be contextual. If the first line of a new `file=` block, *and* the syntax can see that it's a comment in the language that's being used, we know it pretty much *has* to be a shebang. No need for a `header` class there, although that *would* solve the problem.

**Comment by jhidding** [comment]
The first line is not always a comment in the language. For example, in Racket the language is given as

```racket
#lang r6rs

; rest of script
```

While comments start with `;`

An alternative would be to have the user specify a list of regexes in the config for lines that should go to the top.

**Comment by MyriaCore** [comment]
That makes the most sense. We could treat shebang detection the same way we treat language detection and comment detection.

**Comment by jhidding** [comment]
Feature is implemented with version 1.3.0
